### DISCUSSION TITLE
he memory block 'Elsa.Expressions.Models.MemoryBlockReference' does not exist error.

### DISCUSSION THREAD

**Discussion by galgodage** [elsa-workflows/elsa-core#4913] [General]
I execute workflow from json . Here is json.
{
"id": "48f3ccdbc8e7a9c7",
"definitionId": "787d88475a179b36",
"name": "Child Workflow",
"createdAt": "2024-02-04T19:22:54.0083211+00:00",
"version": 1,
"toolVersion": "3.0.0.0",
"variables": [],
"inputs": [
{
"uiHint": "single-line",
"storageDriverType": "Elsa.Workflows.Services.WorkflowStorageDriver, Elsa.Workflows.Core",
"type": "Object",
"name": "test",
"displayName": "test",
"category": "Primitives",
"isArray": false
}
],
"outputs": [],
"outcomes": [],
"customProperties": {
"Elsa:WorkflowContextProviderTypes": []
},
"isReadonly": false,
"isLatest": true,
"isPublished": false,
"options": {
"autoUpdateConsumingWorkflows": false
},
"root": {
"type": "Elsa.Flowchart",
"version": 1,
"id": "477edcca49777c8f",
"nodeId": "Workflow1:477edcca49777c8f",
"metadata": {},
"customProperties": {
"source": "FlowchartJsonConverter.cs:45",
"notFoundConnections": [],
"canStartWorkflow": false,
"runAsynchronously": false
},
"activities": [

{
"text": {
"typeName": "String",
"expression": {
"type": "JavaScript",
"value": "getInput('test')"
},

"memoryReference": {
"id": "WriteLine1:test"
}
},
"id": "bc9e6ebb7c7b3f74",
"nodeId": "Workflow1:477edcca49777c8f:bc9e6ebb7c7b3f74",
"name": "WriteLine1",
"type": "Elsa.WriteLine",
"version": 1,
"customProperties": {
"canStartWorkflow": false,
"runAsynchronously": false
},
"metadata": {
"designer": {
"position": {
"x": -780,
"y": 679.8888893127441
},
"size": {
"width": 139.296875,
"height": 50
}
}
}
}
],
"connections": [
{
"source": {
"activity": "595ac7b8405a0957",
"port": "Done"
},
"target": {
"activity": "bc9e6ebb7c7b3f74",
"port": "In"
}
}
]
}
}

I think error comming from this block

"expression": {
"type": "JavaScript",
"value": "getInput('test')"
},

"memoryReference": {
"id": "WriteLine1:test"
}
this retun The memory block 'Elsa.Expressions.Models.MemoryBlockReference' does not exist error.
Please my deadline is passing. give help to solve this

**Comment by glime-ai** [comment]
Based on the JSON you've provided and the error message, it seems like there might be an issue with the way the `getInput` function is being called or how the memory reference is set up in your workflow definition.

In Elsa, when you want to access an input within an activity, you typically use the context to get the input by its name. However, the `getInput` function is not a standard function provided by Elsa out of the box. If it's a custom function you've defined, you need to ensure it's correctly implemented and accessible within the workflow's execution context.

Additionally, the `memoryReference` seems to be pointing to an ID `"WriteLine1:test"` which might not be correctly set up or might not exist in the workflow definition.

To troubleshoot this issue, you can:

1. Verify that the `getInput` function is correctly defined and accessible within the workflow's JavaScript execution context.
2. Check that the memory reference ID used in the `memoryReference` field corresponds to an actual input or variable within the workflow.
3. Ensure that the input `"test"` is being passed correctly to the workflow when it's being executed.

If you're using Elsa 3.0, you might want to refer to the official documentation or the community support channels for more specific guidance on how to access inputs within a JavaScript expression.

Unfortunately, without access to the actual workflow environment and the custom code, it's challenging to provide a precise solution. If you continue to experience issues, consider reaching out to the Elsa community on GitHub or their discussion forums with detailed information about your workflow and the custom `getInput` function.

If you have any more details or code snippets related to the `getInput` function or how you're executing the workflow, please provide them so I can assist you further.
